### PR TITLE
Allow configuration of variables not currently supported

### PR DESCRIPTION
**Pull request by solomongifford**
This is the same as PR #31 but that repo was moved.

### CODE REVIEW

**Comment by solomongifford** [comment]
I'm actually going to modify this PR - don't pull yet.

**Comment by geerlingguy** [comment]
Sounds good.

**Comment by solomongifford** [comment]
OK, this updated PR goes in a completely different direction and solves another problem as well.

Mysql implementations support using the includedir directive which allows files to override settings in in the my.cnf.  Mariadb does this by default.  

This PR allows the files in the default /etc/mysql/conf.d (Debian based) or /etc/my.cnf.d (CentOS based) to be managed by this ansible role.  Instead of managing the contents of the files, it allows the override files to be managed by the role.   

This PR both solves the need to support more options (as they can be simply added to an override file) and the need to manage the mariadb cnf files.

File: README.md [hunk 1]
```diff
@@ -32,6 +32,10 @@ Whether MySQL should be enabled on startup.
 
 Whether the global my.cnf should be overwritten each time this role is run. Setting this to `no` tells Ansible to only create the `my.cnf` file if it doesn't exist. This should be left at its default value (`yes`) if you'd like to use this role's variables to configure MySQL.
 
+    mysql_config_include_files: []
+
+A list of files that should override the default global my.cnf.   Each item in the array requires a "src" parameter which is a path to a file.  An optional "force" parameter can force the file to be updated each time ansible runs.
```
Comment (geerlingguy) [inline README.md]: Can you change to one space between period/new sentence?

File: defaults/main.yml [hunk 2]
```diff
@@ -70,6 +70,13 @@ mysql_log: ""
 mysql_log_error: /var/log/mysql.err
 mysql_syslog_tag: mysql
 
+mysql_config_include_files: []
+# Full example:
```
Comment (geerlingguy) [inline defaults/main.yml]: Just to follow the example set in many of my other roles, you can delete this line and the following, and just have the following two lines (with the actual example) commented, below the empty (`[]`) var.

File: defaults/main.yml [hunk 3]
```diff
@@ -70,6 +70,13 @@ mysql_log: ""
...
+# mysql_config_include_files:
+#   - src: path/relative/to/playbook/file.cnf
+#   - { src: path/relative/to/playbook/anotherfile.cnf, force: yes }
+
```
Comment (geerlingguy) [inline defaults/main.yml]: Just one space.

File: tasks/configure.yml [hunk 4]
```diff
@@ -9,6 +9,26 @@
...
+  when: mysql_config_include_files|length
+
+- name: Copy my.cnf override files into include directory.
+  template:
+    src: "{{ item.src }}"
+    dest: "{{ mysql_config_include_dir }}/{{ item.src | basename }}"
+    owner: root
+    group: root
+    mode: 0644
+    force: "{{ item.force|default(False) }}"
```
Comment (geerlingguy) [inline tasks/configure.yml]: please include a space between the brackets.

File: templates/my.cnf.j2 [hunk 5]
```diff
@@ -96,3 +96,10 @@ max_allowed_packet = {{ mysql_mysqldump_max_allowed_packet }}
 
 [mysqld_safe]
 pid-file = {{ mysql_pid_file }}
+
+{% if mysql_config_include_files|length %}
+# * IMPORTANT: Additional settings that can override those from this file!
+#   The files must end with '.cnf', otherwise they'll be ignored.
+#
+!includedir {{ mysql_config_include_dir }}
+{% endif %}
```
Comment (geerlingguy) [inline templates/my.cnf.j2]: needs newline

File: templates/my.cnf.j2 [hunk 6]
```diff
@@ -96,3 +96,10 @@ max_allowed_packet = {{ mysql_mysqldump_max_allowed_packet }}
 
 [mysqld_safe]
 pid-file = {{ mysql_pid_file }}
+
+{% if mysql_config_include_files|length %}
```
Comment (geerlingguy) [inline templates/my.cnf.j2]: Please add space around |

**Comment by geerlingguy** [comment]
I really like the approach; it offers as much flexibility as the role consumer needs, and is much nicer than blindly adding hundreds more options to the config.

File: defaults/main.yml (hunk 2 above)
Comment (solomongifford) [inline defaults/main.yml]: Sure - I assume you'll want the other examples in this cleaned up as well, but I'll leave that for another day.

**Comment by solomongifford** [comment]
Let me know if I caught all the spacing changes.

**Comment by geerlingguy** [comment]
@solomongifford - ah, just noticed the others; I've fixed them in a follow-up commit. Thanks!

**Pull request merged**